src/matrix_utilities.jl

```diff
@@ -10,29 +10,30 @@ end
 lsqrt(m::AbstractMatrix) = cholesky(m).L
 lsqrt(m::UniformScaling) = sqrt(m)
 
-# stein operator
-stein(Σ, Φ) = Matrix(Hermitian(Φ * Σ * Φ'))
-stein(Σ::T, Φ::T) where {T<:UniformScaling} = Φ * Σ * Φ'
+# project matrix onto symmetric matrix
+symmetrise(Σ) = Σ
+symmetrise(Σ::Matrix{T}) where {T} = T <: Real ? Symmetric(Σ) : Hermitian(Σ)
```

Comment:
This is unintuitive.. For example, `symmetrise(Tridiagonal(1*ones(1), 2*ones(2), 3*ones(1)))` does not return a symmetric matrix.

What is the issue with just doing:
```suggestion
symmetrise(Σ::AbstractMatrix{T}) where {T} = T <: Real ? Symmetric(Σ) : Hermitian(Σ)
```

Reply:
Ah you want to cover the `UniformScaling` case I guess? Then just add
`symmetrise(Σ::UniformScaling) = Σ` or something?

Reply:
Verdict: Wrap by default; dispatch symmetrise on custom types to not do anything if it should not do anything.